### PR TITLE
Fix `performanceFlags` and make 16bpp explicit

### PR DESCRIPTION
**Pull request by ibeckermayer**
- Adds more specificity for detecting the RDP version in `pub enum Version`
- Fixes a bug in `impl From<u32> for Version`
- Fixes a bug in `rdp_extended_infos` which was preventing us from being able to send performance flags
- Request 16bpp explicitly rather than requesting 24bpp but also saying we only support 16bpp or 32bpp

### CODE REVIEW

File: src/core/sec.rs [hunk 1]
```diff
@@ -74,9 +74,9 @@ enum AfInet {
 fn rdp_extended_infos(performance_flags: u32) -> Component {
     component![
         "clientAddressFamily" => U16::LE(AfInet::AfInet as u16),
-        "cbClientAddress" => DynOption::new(U16::LE(0), |x| MessageOption::Size("clientAddress".to_string(), x.inner() as usize + 2)),
+        "cbClientAddress" => U16::LE(2),
```
Comment (zmb3) [inline src/core/sec.rs:78]: What exactly was wrong here?

File: src/core/sec.rs [hunk 2]
```diff
@@ -74,9 +74,9 @@ enum AfInet {
...
         "clientAddress" => b"\x00\x00".to_vec(),
-        "cbClientDir" => U16::LE(0),
+        "cbClientDir" => U16::LE(2),
```
Comment (zmb3) [inline src/core/sec.rs:80]: What's the 2? I think any time we use something that's non-zero we should use a constant or link to the spec so it's easy to look up.

**Review by zmb3 — COMMENTED** [review]
How did you fix the `SEC: Invalid Licence packet` issue?

File: src/core/sec.rs (hunk 1 above)
Comment (ibeckermayer) [reply src/core/sec.rs:78]: I'm not exactly sure what was meant to happen there, but it seems like it was something like: calculate the size of the string in the `"clientAddress"` field below and add 2 for the two null-terminator bytes.

File: src/core/sec.rs (hunk 2 above)
Comment (ibeckermayer) [reply src/core/sec.rs]: 6df9004f2d4fbaafa0aebcfa98c70cd9cb2554a5

**Comment by ibeckermayer** [comment]
> How did you fix the SEC: Invalid Licence packet issue?

I tried these changes under the assumption that that was caused by an "off-by-one" type of error. Now that I've fixed this little section, it no longer thinks it's trying to parse any sort of license packet. (I'm not sure precisely what part of the protocol that was coming from).

@zmb3

File: src/core/sec.rs (hunk 1 above)
Comment (zmb3) [reply src/core/sec.rs]: I feel like I may have written this to fix a security issue. I'll check the git blame tomorrow.

Comment (ibeckermayer) [reply src/core/sec.rs]: I just checked -- no, it was citronneur. You only wrote the line
```rust
"performanceFlags" => U32::LE(performance_flags)
```

I don't see any potential security bugs here, we're just passing hardcoded flags to the RDP server.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**